Add note regarding `\u{2011}` logging a `genstrings` warning

mokagio opened the pull request:
### Description
What is says in the title. See https://github.com/woocommerce/woocommerce-ios/issues/8219 for more details.

While I was at it, I adde reverse-DNS keys to all the localized strings in the context of the annotated one.

### Testing instructions
N.A.

---
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

<!-- Pull request guidelines: https://github.com/woocommerce/woocommerce-android/blob/develop/docs/pull-request-guidelines.md -->

cc @iamgabrielma as the strings author. Just to be clear, I think you made the right call, it's just unfortunate the tooling does this to us.

mokagio committed:
Use reverse-DNS keys in `CardPresentModalScanningForReader` strings

mokagio committed:
Add note regarding `\u{2011}` logging a `genstrings` warning

AliSoftware commented on WooCommerce/Classes/ViewModels/CardPresentPayments/CardPresentModalScanningForReader.swift:
```diff
@@ -81,30 +81,45 @@ private extension CardPresentModalScanningForReader {
                      """
         )
 
+        // \u{2011} (Non-Breaking Hyphen) will result in `genstrings` logging
+        //
+        // invalid unicode sequence: \u{201
+        //
+        // This is a known bug in the toolchain.
+        // The only known workaround is to use the actual character, but that would defeat the non-breaking purpose.
```
For the record, I double-checked if using the `-u` flag of `genstrings` would help fix that issue (despite the man page saying that this flag is more to "Allow unicode characters in the value of strings files." rather than about parsing Unicode characters in source code) and confirmed it didn't make the issue go away.

> The only known workaround is to use the actual character, but that would defeat the non-breaking purpose.

I'm not sure I'm following what would be so wrong about using the actual "Non-Breaking Hyphen" literal character in the source code directly 🤔 
Is it to make sure that the translators also use a non-breaking hyphen in their translations (as in: there's more chances that they keep the `\u{2011}` verbatim in their translation, compared to if we used a `‑` literal in the source copy and let them re-type the `In‑Person` text as part of their translation)?

But I'm guessing even the `In‑Person` text will be translated in something else in the other locales anyway, so each locale should use the appropriate word for it and that word might not contain a hyphen in other locales to begin with…

If that's so, I'm not sure I agree with the rationale _not_ to use `‑` literal in the first place, and I think we might be better off using `‑` literal in the `value:`, then replace the `\u{2011} is a special character that acts as nonbreaking hyphen for "-" in the "In-Person" string.` mention in the `comment:` with something like `The hyphen in the "In‑Person" string is a non-breaking hyphen (U+2011). If your translation of that term also happens to contains a hyphen, please be sure to use a non-breaking hyphen character for it too.`

mokagio replied:
@iamgabrielma keen to hear what you think about this and if there's a strong reason use `U+2011`.

iamgabrielma replied:
Thanks for the ping! As I recall the reason behind using the unicode there was mostly design, as in some of small screens this word was cut off when the alert was presented, (or showing entirely on a new line) and we generally use "In-Person" together all across our public [references]( https://woocommerce.com/in-person-payments/). 

That said, this only may appear to a small subset of users, for a very specific modal, during a couple of seconds, and the message will still be still perfectly understandable, so I agree with Olivier and I think we can just switch to use the hyphen literal directly if this is causing warnings. We can just clarify it in the comment for translators 👍

AliSoftware replied:
@iamgabrielma To be clear, I'm not advocating we replace the non-breaking-hyphen with a regular hyphen. I'm suggesting we type the non-breaking-hyphen literal directly in the code, verbatim, instead of representing it using `\u{2011}`.

The end result would be completely unchanged, since the `String` would still contain a non-breaking-hyphen literal just like before.
It's just that by avoiding to use the `\u{2011}` Swift escape code for that character, and instead using the literal for non-breaking-hyphen directly, we will avoid encountering the bug with `genstrings` that doesn't seem to like `\u{` when parsing the code for `NSLocalizedString`.

iamgabrielma replied:
Gotcha! I miss-understood on my first reading. That is even better, thanks! 🙇

mokagio committed:
Replace escaped `U+2011` with actual character in a localized string

Hat tip @AliSoftware

See conversation at
https://github.com/woocommerce/woocommerce-ios/pull/8220#discussion_r1032414275

mokagio commented on WooCommerce/Classes/ViewModels/CardPresentPayments/CardPresentModalScanningForReader.swift:
```diff
@@ -107,22 +107,13 @@ private extension CardPresentModalScanningForReader {
                      A label prompting users to learn more about In-Person Payments.
-                     \u{2011} is a special character that acts as nonbreaking hyphen for "-" in the "In-Person" string.
+                     The hyphen in "In‑Person" is a non-breaking hyphen (U+2011).
+                     If your translation of that term also happens to contains a hyphen, please be sure to use the non-breaking hyphen character for it.
```
With both `\u{2011}` removed, `genstrings` no longer prints warnings

![image](https://user-images.githubusercontent.com/1218433/204548278-25304ca5-7ee9-4232-8c88-10f9f8a31055.png)
